crates/sui-framework-snapshot/src/lib.rs

```diff
@@ -1,8 +1,53 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-use std::{io::Read, path::PathBuf};
+use serde::{Deserialize, Serialize};
+use std::collections::BTreeMap;
+use std::{fs, io::Read, path::PathBuf};
 use sui_framework::SystemPackage;
+use sui_types::base_types::ObjectID;
+
+pub type SnapshotManifest = BTreeMap<String, BTreeMap<u64, SingleSnapshot>>;
+
+#[derive(Serialize, Deserialize)]
+pub struct SingleSnapshot {
+    /// Git revision that this snapshot is taken on.
+    git_revision: String,
+    /// List of file names (also identical to object ID) of the bytecode package files.
+    package_ids: Vec<ObjectID>,
+}
+
+pub fn load_bytecode_snapshot_manifest() -> SnapshotManifest {
+    let Ok(bytes) = fs::read(manifest_path()) else {
+        return SnapshotManifest::default();
+    };
+    serde_json::from_slice::<SnapshotManifest>(&bytes)
+        .expect("Could not deserialize SnapshotManifest")
+}
+
+pub fn update_bytecode_snapshot_manifest(
+    network: &str,
+    git_revision: String,
+    version: u64,
```

Comment:
This version will almost always be the max version right? unless we want to update the format (but not the contents)of some previous snapshots

Reply:
That's correct.